src/DocBlock/TypeExpression.php

```diff
@@ -208,8 +208,9 @@ final class TypeExpression
 
     private string $value;
 
-    private bool $isUnionType;
+    private bool $isGluedType;
```

Comment:
```suggestion
    private bool $isCompositeType;
```

Actually I [worked on similar improvement](https://github.com/Wirone/PHP-CS-Fixer/commit/cc4ce62a75760d319743dbdcbb67582cc1a72898) around this, because it was confusing me every time I saw the usage 😅. I will leave some feedback based on what I personally did here.

Reply:
reverted completely because of `~` not implemented in this PR - https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/8210#discussion_r1765400782